x/pool-incentives/keeper/grpc_query_test.go

```diff
@@ -35,89 +30,116 @@ func (suite *KeeperTestSuite) TestGaugeIds() {
+			if tc.err {
+				suite.Require().Error(err)
+			} else {
+				suite.Require().NoError(err)
+				suite.NoError(err)
+				suite.Equal(3, len(res.GaugeIdsWithDuration))
+
+				poolLpDenom := gammtypes.GetPoolShareDenom(pool.GetId())
+				for i := 0; i < len(res.GaugeIdsWithDuration); i++ {
+					suite.Equal(lockableDurations[i], res.GaugeIdsWithDuration[i].Duration)
+
+					gauge, err := suite.App.IncentivesKeeper.GetGaugeByID(suite.Ctx, res.GaugeIdsWithDuration[i].GaugeId)
+					suite.NoError(err)
+					suite.Equal(0, len(gauge.Coins))
+					suite.Equal(true, gauge.IsPerpetual)
+					suite.Equal(poolLpDenom, gauge.DistributeTo.Denom)
+					suite.Equal(lockableDurations[i], gauge.DistributeTo.Duration)
+				}
+			}
+		})
+	}
 }
 
 func (suite *KeeperTestSuite) TestDistrInfo() {
 	suite.SetupTest()
 
-	queryClient := suite.queryClient
-
-	res, err := queryClient.DistrInfo(context.Background(), &types.QueryDistrInfoRequest{})
-	suite.NoError(err)
-
-	suite.Equal("0", res.DistrInfo.TotalWeight.String())
-	suite.Equal(0, len(res.DistrInfo.Records))
-}
-
-func (suite *KeeperTestSuite) TestDistrInfo2() {
-	suite.SetupTest()
-
 	keeper := suite.App.PoolIncentivesKeeper
 	queryClient := suite.queryClient
 
-	poolId := suite.PrepareBalancerPool()
-
-	// LockableDurations should be 1, 3, 7 hours from the default genesis state.
-	lockableDurations := keeper.GetLockableDurations(suite.Ctx)
-	suite.Equal(3, len(lockableDurations))
-
-	gauge1Id, err := keeper.GetPoolGaugeId(suite.Ctx, poolId, lockableDurations[0])
-	suite.NoError(err)
-
-	gauge2Id, err := keeper.GetPoolGaugeId(suite.Ctx, poolId, lockableDurations[1])
-	suite.NoError(err)
-
-	gauge3Id, err := keeper.GetPoolGaugeId(suite.Ctx, poolId, lockableDurations[2])
-	suite.NoError(err)
-
-	// Create 3 records
-	err = keeper.UpdateDistrRecords(suite.Ctx, types.DistrRecord{
-		GaugeId: gauge1Id,
-		Weight:  sdk.NewInt(100),
-	}, types.DistrRecord{
-		GaugeId: gauge2Id,
-		Weight:  sdk.NewInt(200),
-	}, types.DistrRecord{
-		GaugeId: gauge3Id,
-		Weight:  sdk.NewInt(300),
-	})
-	suite.NoError(err)
-
-	res, err := queryClient.DistrInfo(context.Background(), &types.QueryDistrInfoRequest{})
-	suite.NoError(err)
-
-	suite.Equal("600", res.DistrInfo.TotalWeight.String())
-	suite.Equal(3, len(res.DistrInfo.Records))
+	for _, tc := range []struct {
+		desc                 string
+		poolCreated          bool
+		weights              []sdk.Int
+		expectedTotalWeight  sdk.Int
+		expectedRecordLength int
+	}{
+		{
+			desc:                 "No pool exist",
+			poolCreated:          false,
+			weights:              []sdk.Int{},
+			expectedTotalWeight:  sdk.NewInt(0),
+			expectedRecordLength: 0,
+		},
+		{
+			desc:                 "Happy case",
+			poolCreated:          true,
+			weights:              []sdk.Int{sdk.NewInt(100), sdk.NewInt(200), sdk.NewInt(300)},
+			expectedTotalWeight:  sdk.NewInt(600),
+			expectedRecordLength: 3,
+		},
+	} {
+		tc := tc
+		suite.Run(tc.desc, func() {
+			if tc.poolCreated {
```

Comment:
is `poolCreated` = false the same as `len(weights).= 0` 

We can have a valid pool without any tokens locked right?